src/devices/machine/s2350.h

```diff
@@ -0,0 +1,121 @@
+// license:BSD-3-Clause
+// copyright-holders:Mark Garlanger
+/**********************************************************************
+
+ American Systems Inc. S2350 Universal Synchronous Receiver/Transmitter
+
+**********************************************************************/
+
+#ifndef MAME_MACHINE_S2350_H
+#define MAME_MACHINE_S2350_H
+
+#pragma once
+
+#include "machine/timer.h"
+
+
+//
+// AMI S2350 USRT
+//
+class s2350_device : public device_t
+{
+public:
+	s2350_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	void rx_w(int state) { serial_rx_w(state); }
+
+	auto tx_handler() { return m_tx_cb.bind(); }
+
+	void transmitter_holding_reg_w(uint8_t data);
+	void transmit_fill_reg_w(uint8_t data);
+	void receiver_sync_reg_w(uint8_t data);
+
+	u8   receiver_output_reg_r();
+	u8   status_word_r();
+	u8   receiver_sync_search();
+
+	// transmit statuses
+	auto transmit_buffer_empty_cb()      { return m_tbmt.bind(); }
+	auto fill_character_transmitted_cb() { return m_fct.bind(); }
+
+	// receive statuses
+	auto received_data_available_cb()    { return m_rda.bind(); }
+	auto sync_character_received_cb()    { return m_scr.bind(); }
+	auto receiver_overrun_cb()           { return m_ror.bind(); }
+	auto receiver_parity_error_cb()      { return m_rpe.bind(); }
+
+	void receiver_restart();
+
+	// transmitter clock
+	void tcp_w();
+	// receiver clock
+	void rcp_w();
+
+protected:
+
+	virtual void device_start() override;
+	virtual void device_reset() override;
+
+	void serial_rx_w(int state);
+	void update_receiver_shift();
+
+	void set_transmit_buffer_empty(int val);
+	void set_fill_char_transmitted(int val);
+
+	void set_received_data_available(int val);
+	void set_receiver_overrun(int val);
+	void set_receiver_parity_error(int val);
+	void set_sync_character_received(int val);
```

Comment:
Can these use `bool` parameters?  They aren’t input line handler functions, so they don’t need to use the standard input line handler function signature.